src/test/java/io/cdap/plugin/orientdb/OrientDBConfigTest.java

```diff
@@ -0,0 +1,137 @@
+    Assert.assertTrue(failureCollector.getValidationFailures().isEmpty());
+  }
+
+  @Test
+  public void testInvalidReferenceName() {
+    OrientDBConfig config = OrientDBConfig.builder(VALID_CONFIG)
+      .setReferenceName("!@#$%^&*()")
+      .build();
+
+    MockFailureCollector failureCollector = new MockFailureCollector(MOCK_STAGE);
+    config.validate(failureCollector, VALID_SCHEMA);
+    remoteDatabaseConnectionFailure(failureCollector);
+    assertValidationFailed(failureCollector, OrientDBConfig.REFERENCE_NAME);
+  }
+
+  @Test
+  public void testVertexAbsent() {
+    OrientDBConfig config = OrientDBConfig.builder(VALID_CONFIG)
+      .setVertexType("nonExistingField")
+      .build();
+
+    MockFailureCollector failureCollector = new MockFailureCollector(MOCK_STAGE);
+    config.validate(failureCollector, VALID_SCHEMA);
+    remoteDatabaseConnectionFailure(failureCollector);
+    assertValidationFailed(failureCollector, OrientDBConfig.VERTEX);
+  }
+
+  @Test
+  public void testEdgeAbsent() {
+    OrientDBConfig config = OrientDBConfig.builder(VALID_CONFIG)
+      .setEdgeType("nonExistingField")
+      .build();
+
+    MockFailureCollector failureCollector = new MockFailureCollector(MOCK_STAGE);
+    config.validate(failureCollector, VALID_SCHEMA);
+    remoteDatabaseConnectionFailure(failureCollector);
+    assertValidationFailed(failureCollector, OrientDBConfig.EDGE);
+  }
+
+  @Test
+  public void testVertexInvalidType() {
+    Schema schema =
+      Schema.recordOf("schema",
+                      Schema.Field.of("MyV", Schema.nullableOf(Schema.of(Schema.Type.BYTES))),
+                      Schema.Field.of("MyE",
+                                      Schema.nullableOf(Schema.arrayOf(Schema.nullableOf(Schema.of(Schema.Type.STRING
+                                      ))))));
+
+    MockFailureCollector failureCollector = new MockFailureCollector(MOCK_STAGE);
+    VALID_CONFIG.validate(failureCollector, schema);
+    remoteDatabaseConnectionFailure(failureCollector);
+    assertValidationFailed(failureCollector, OrientDBConfig.VERTEX);
+  }
+
+  @Test
+  public void testEdgeInvalidType() {
+    Schema schema =
+      Schema.recordOf("schema",
+                      Schema.Field.of("MyV", Schema.nullableOf(Schema.of(Schema.Type.STRING))),
+                      Schema.Field.of("MyE",
+                                      Schema.nullableOf(Schema.arrayOf(Schema.nullableOf(Schema.of(Schema.Type.BYTES
+                                      ))))));
+
+    MockFailureCollector failureCollector = new MockFailureCollector(MOCK_STAGE);
+    VALID_CONFIG.validate(failureCollector, schema);
+    remoteDatabaseConnectionFailure(failureCollector);
+    assertValidationFailed(failureCollector, OrientDBConfig.EDGE);
+  }
+
+  private static void remoteDatabaseConnectionFailure(MockFailureCollector failureCollector) {
+    Iterator<ValidationFailure> failuresIterator = failureCollector.getValidationFailures().iterator();
```

Comment:
1. probably it is not the best idea to connect to DB from config validation method
2. definitely a bad idea to remove validation failure this way in tests. Very dirty workaround.